`Assessment`: Fix translation error when deleting manual feedback

b-fein opened the pull request:
<!-- Thanks for contributing to Artemis! Before you submit your pull request, please make sure to check the following boxes by putting an x in the [ ] (don't: [x ], [ x], do: [x]) -->
<!-- If your pull request is not ready for review yet, create a draft pull request! -->

### Checklist
#### General
<!-- You only need to choose one of the first two check items: Generally, test on the test servers. -->
<!-- If it's only a small change, testing it locally is acceptable and you may remove the first checkmark. If you are unsure, please test on the test servers. -->
- [ ] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).
#### Client
- [x] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data.
- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] Following the [theming guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-design/), I specified colours only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [x] I translated all newly inserted strings into English and German.

### Motivation and Context
The translation when deleting manual feedback during assessment sometimes showed `{{ text }}` instead of the actual feedback text (see screenshot below).

### Description
When using grading instructions, the instruction already provides some text and the feedback detail text is allowed to remain empty. If the text is empty, we now add the grading instruction text to the dialogue.
In case some manual detail text is present, we still use that.

### Steps for Testing
<!-- Please describe in detail how the reviewer can test your changes. -->
Prerequisites:
- 1 Instructor
- 1 Exercise
  - with manual assessment enabled
  - grading instructions (you can add them in the exercise settings)
  - $\geqq 1$ submissions that can be assessed

1. Start the manual assessment.
2. Create a feedback from a grading instruction by dragging the instruction from the right into a feedback (either at the bottom for general feedback, or in case of programming exercises also for inline feedback).
3. Do *not* add additional manual detail text to the feedback.
4. If inline code feedback: Save this feedback.
5. Delete the feedback. You should get a dialogue as shown in the screenshot, but it shows the grading instruction text instead of `{{ text }}`.

### Testserver States
> [!NOTE]
> These badges show the state of the test servers.
> Green = Currently available, Red = Currently locked

![](https://byob.yarr.is/ls1intum/Artemis/artemis-test1)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test2)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test3)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test4)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test5)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test6)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test9)

### Review Progress
<!-- Each Pull Request should be reviewed by at least two other developers. The code, the functionality (= manual test) and the exam mode need to be reviewed. -->
<!-- The reviewer or author check the following boxes depending on what was reviewed or tested. All boxes should be checked before merge. -->
<!-- You can add additional checkboxes if it makes sense to only review parts of the code or functionality. -->
<!-- When changes are pushed, uncheck the affected boxes. (Not all changes require full re-reviews.) -->
<!-- All PRs that might affect the exam mode (e.g. change a client component that is also used in the exam mode) need an additional verification that the exam mode still works. -->

#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Test Coverage
| Class | Coverage | Assertions |
| --- | --- | --- |
| FeedbackContentPipe | 100% | :heavy_check_mark: |
| QuotePipe | 100% | :heavy_check_mark: |

### Screenshots
before:
![grafik](https://github.com/ls1intum/Artemis/assets/64250573/83dfc875-4063-4125-804e-623b8c563a87)

@coderabbitai ignore

konrad2002 reviewed (changes requested):
tested on legacy ts11. If you drag and drop the assessment instruction into the feedback area and delete feedback without description, it works as described.

However, I found two issues:

1. when assessing a line in the code, adding feedback without description, and deleting it later, the message still includes `{{ text }}`.
2. when you drag and drop an assessment instruction into an already existing feedback, remove the old description, override the assessment and delete it, no description is shown.
![grafik](https://github.com/ls1intum/Artemis/assets/44098768/b76b3173-c1d4-4c93-9d42-ad54379925ab)

konrad2002 reviewed (approved):
tested again on legacy ts11, everything works fine now

LenaKahle reviewed (commented):
Tried testing on ts4 and legacy ts11, both times the drag-and-dropped feedback without additional text can't be saved and when deleting it the prompt is just "Are you sure you want to delete the feedback?" and doesn't contain any text. (When I manually enter text that text is displayed in the dialogue window)
Is there a way to save the feedback without content?

b-fein commented:
> drag-and-dropped feedback without additional text can't be saved and when deleting it the prompt is just "Are you sure you want to delete the feedback?"

@LenaKahle Sorry about that, I updated the steps for testing. Saving only works for inline code feedback (there is no delete available otherwise). For the general feedback at the bottom the wastebin icon can be used directly without saving.

chrisknedl reviewed (approved):
Code looks good

LenaKahle reviewed (dismissed):
Approve, tested on ts4.

RY997 reviewed (dismissed):
Code LGTM

Strohgelaender reviewed (dismissed):
code

rabeatwork commented on src/main/webapp/app/assessment/unreferenced-feedback-detail/unreferenced-feedback-detail.component.html:
```diff
@@ -10,7 +10,7 @@
                 jhiDeleteButton
                 [buttonSize]="ButtonSize.SMALL"
-                [translateValues]="{ text: feedback.detailText }"
+                [translateValues]="{ text: Feedback.hasContent(feedback) ? ' \'' + Feedback.getContent(feedback) + '\'' : '' }"
```
Please avoid calling methods from your html template. Instead, you should calculate the desired values beforehand and assign them to a variable.

b-fein replied:
Best I can do is put the value in a `get` attribute (see new commit). The user can still change the feedback text and the grading instruction.

rabeatwork commented on src/main/webapp/app/exercises/programming/assess/code-editor-tutor-assessment-inline-feedback.component.html:
```diff
@@ -66,7 +66,7 @@
                         jhiDeleteButton
                         [buttonSize]="ButtonSize.MEDIUM"
-                        [translateValues]="{ text: feedback.detailText }"
+                        [translateValues]="{ text: Feedback.hasContent(feedback) ? ' \'' + Feedback.getContent(feedback) + '\'' : '' }"
```
Same here

nata-81 reviewed (changes requested):
Tested on ts2 with a modeling exercise. The text translation works as expected but there is a missing space between the word "feedback" and the grading text. See screenshot: 
<img width="595" alt="image" src="https://github.com/ls1intum/Artemis/assets/67416888/5ec70690-41fb-49cc-92b8-cc8238e8243c">

nata-81 reviewed (approved):
Tested on ts2 works as expected. Except, if there are multiple feedbacks there is a chance that some of them won't show the grading text at all (BUT I DON'T KNOW IF THIS IS RELATED TO THIS PR). See screenshot 1 the first two show no grading text like in picture 2 and the third and the fourth show a grading text like in picture. 
Picture 1: 
<img width="624" alt="image" src="https://github.com/ls1intum/Artemis/assets/67416888/e5015d7c-b6aa-4a2d-aa70-b09c7cd68434">
 picture 2:
<img width="584" alt="image" src="https://github.com/ls1intum/Artemis/assets/67416888/f5767062-d250-4afd-a795-c47d6a56dcf4">
picture 3:
<img width="596" alt="image" src="https://github.com/ls1intum/Artemis/assets/67416888/d5186c29-3625-4bfc-9ab6-21034765a2d5">

Gusti2010 reviewed (approved):
Tested with a text exercise on t11_Legacy and it works as described. For inline feedback in a test exercise, no delete window is shown at all. However, your test steps specify that we should use inline feedback only for programming exercises, so i think this is fine!
For general feedback it works as expected

georgyia reviewed (approved):
manually tested on ts2, lgtm

Predixx reviewed (approved):
tested and works.
![grafik](https://github.com/ls1intum/Artemis/assets/63974342/50dea372-05e3-44f4-b1a0-a81bdfda3f97)

chrisknedl reviewed (approved):
Code looks good
